src/Renderer/ThreeExtended/OBB.js

```diff
@@ -1,133 +1,168 @@
+        this.box3D.max.z = this.natBox.max.z + max;
+        this.update();
+    }
 
-// get oriented bounding box of tile
-const builder = new BuilderEllipsoidTile();
-OBB.extentToOBB = function _extentToOBB(extent, minHeight = 0, maxHeight = 0) {
-    if (extent._crs != 'EPSG:4326') {
-        throw new Error('The extent crs is not a Geographic Coordinates (EPSG:4326)');
+    /**
+     * Set bouding box value to points
+     *
+     * @param      {Array<THREE.Vector3>}  points  The points to set
+     * @return     {Array<THREE.Vector3>}  The points seted
+     */
+    toPoints(points) {
+        // top points of bounding box
+        points[0].set(this.box3D.max.x, this.box3D.max.y, this.box3D.max.z);
+        points[1].set(this.box3D.min.x, this.box3D.max.y, this.box3D.max.z);
+        points[2].set(this.box3D.min.x, this.box3D.min.y, this.box3D.max.z);
+        points[3].set(this.box3D.max.x, this.box3D.min.y, this.box3D.max.z);
+        // bottom points of bounding box
+        if (points.length > 4) {
+            points[4].set(this.box3D.max.x, this.box3D.max.y, this.box3D.min.z);
+            points[5].set(this.box3D.min.x, this.box3D.max.y, this.box3D.min.z);
+            points[6].set(this.box3D.min.x, this.box3D.min.y, this.box3D.min.z);
+            points[7].set(this.box3D.max.x, this.box3D.min.y, this.box3D.min.z);
+        }
+
+        return points;
     }
 
-    const { sharableExtent, quaternion, position } = builder.computeSharableExtent(extent);
-    const paramsGeometry = {
-        extent: sharableExtent,
-        level: 0,
-        segment: 2,
-        disableSkirt: true,
-    };
-
-    const geometry = new TileGeometry(paramsGeometry, builder);
-    const obb = geometry.OBB;
-    obb.updateZ(minHeight, maxHeight);
-    obb.position.copy(position);
-    obb.quaternion.copy(quaternion);
-    obb.update();
-
-    // Calling geometry.dispose() is not needed since this geometry never gets rendered
-    return obb;
-};
+    /**
+     * Determines if the sphere is above the XY space of the box
+     *
+     * @param      {Sphere}   sphere  The sphere
+     * @return     {boolean}  True if sphere is above the XY space of the box, False otherwise.
+     */
+    isSphereAboveXYBox(sphere) {
+        const localSpherePosition = this.worldToLocal(sphere.position);
+        // get obb closest point to sphere center by clamping
+        const x = Math.max(this.box3D.min.x, Math.min(localSpherePosition.x, this.box3D.max.x));
+        const y = Math.max(this.box3D.min.y, Math.min(localSpherePosition.y, this.box3D.max.y));
+
+        // this is the same as isPointInsideSphere.position
+        const distance = Math.sqrt((x - localSpherePosition.x) * (x - localSpherePosition.x) +
+                               (y - localSpherePosition.y) * (y - localSpherePosition.y));
+
+        return distance < sphere.radius;
+    }
+
+    /**
+     * Compute OBB from extent.
+     * The OBB resulted can be only in the system 'EPSG:3946'.
```

Comment:
Do you mean `4326` ?

Reply:
No, the extent in entry is in `4326` but the result is `3946` because it is meant to viewer